### PR TITLE
game and test

### PR DESCRIPTION
**Pull request by YaffaLevy**

### CODE REVIEW

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package levy.gameoflife;
+
+public class GameOfLife {
+    private int width;
+    private int height;
+    int[][] grid;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java]: private

File: src/test/java/levy/gameoflife/GameOfLifeTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package levy.gameoflife;
+
+import levy.gameoflife.GameOfLife;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class GameOfLifeTest {
+
+    @Test
+    public void blinkerPattern() {
```
Comment (andrewoid) [inline src/test/java/levy/gameoflife/GameOfLifeTest.java]: `nextGen()`

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+                // rules
+                if (grid[i][j] == 1) {
+                    if (liveNeighbors < 2 || liveNeighbors > 3) {
+                        newGrid[i][j] = 0;
+                    } else {
+                        newGrid[i][j] = 1;
+                    }
+                } else {
+                    if (liveNeighbors == 3) {
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java:30]: else-if

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        int[][] newGrid = new int[height][width];
+
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                int liveNeighbors = countLiveNeighbors(i, j);
+
+                // rules
+                if (grid[i][j] == 1) {
+                    if (liveNeighbors < 2 || liveNeighbors > 3) {
+                        newGrid[i][j] = 0;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java:25]: You don't need to set it to 0, it's already 0

File: src/main/java/levy/gameoflife/GameOfLife.java (hunk 1 above)
Comment (YaffaLevy) [reply src/main/java/levy/gameoflife/GameOfLife.java]: when i make private then the grid in my test doesnt have acess - the Ide gave me the option of packaged-private what is that? does that help?

Comment (andrewoid) [reply src/main/java/levy/gameoflife/GameOfLife.java]: Make a method that allows you to pass it. Don't access set the value directly like that.

Comment (YaffaLevy) [reply src/main/java/levy/gameoflife/GameOfLife.java]: just a setter no getter right?

Comment (andrewoid) [reply src/main/java/levy/gameoflife/GameOfLife.java]: Unless you think you need both...

Comment (YaffaLevy) [reply src/main/java/levy/gameoflife/GameOfLife.java]: i mean getter never used so no

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    } else {
+                        newGrid[i][j] = 1;
+                    }
+                } else {
+                    if (liveNeighbors == 3) {
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java]: Else-if

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        int[][] newGrid = new int[height][width];
+
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                int liveNeighbors = countLiveNeighbors(i, j);
+
+                // rules
+                if (grid[i][j] == 1) {
+                    if (liveNeighbors < 2 || liveNeighbors > 3) {
+                        newGrid[i][j] = 0;
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java]: This is already 0

File: src/main/java/levy/gameoflife/GameOfLife.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            for (int j = 0; j < width; j++) {
+                int liveNeighbors = countLiveNeighbors(i, j);
+                //rules
+                if (grid[i][j] == 1) {
+                    if (liveNeighbors == 2 || liveNeighbors == 3) {
+                        newGrid[i][j] = 1;
+                    }
+                } else if (liveNeighbors == 3) {
+                    newGrid[i][j] = 1;
+                }
```
Comment (andrewoid) [inline src/main/java/levy/gameoflife/GameOfLife.java]: You can combine this further.

Comment (YaffaLevy) [reply src/main/java/levy/gameoflife/GameOfLife.java]: still further?

**Review by andrewoid — APPROVED** [review]

**Pull request merged**